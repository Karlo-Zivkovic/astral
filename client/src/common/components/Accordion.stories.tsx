import React from 'react';
import { ComponentStory, ComponentMeta } from '@storybook/react';

import Accordion from './Accordion';

export default {
  title: 'Common/Accordion',
  component: Accordion,
} as ComponentMeta<typeof Accordion>;

const Template: ComponentStory<typeof Accordion> = (args) => (
  <Accordion {...args} />
);

export const Primary = Template.bind({});
Primary.args = {
  title: 'Accordion title',
  children: 'Accordion body',
};

export const Secondary = Template.bind({});
Secondary.args = {
  title: 'Accordion title',
  children: 'Accordion body',
  value: '213',
};
