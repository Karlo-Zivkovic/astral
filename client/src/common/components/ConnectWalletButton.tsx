import { WalletIcon } from '@heroicons/react/24/outline'
import useMediaQuery from 'common/hooks/useMediaQuery'
import React, { useCallback, useState } from 'react'
import PreferredExtensionModal from '../../layout/components/PreferredExtensionModal'

export const ConnectWalletButton: React.FC = () => {
  const [walletModalIsOpen, setWalletModalIsOpen] = useState(false)
  const isDesktop = useMediaQuery('(min-width: 1024px)')

  const handleConnectWallet = useCallback((e: React.MouseEvent<HTMLButtonElement, MouseEvent>) => {
    e.preventDefault()
    setWalletModalIsOpen(true)
  }, [])
  const handleWalletModalOnClose = useCallback(() => setWalletModalIsOpen(false), [])

  return (
    <>
      <button
        onClick={handleConnectWallet}
        className={`h-10 ${
          isDesktop ? 'w-36' : 'w-10 py-2 px-2'
        } text-white font-medium bg-gradient-to-r from-[#EA71F9] to-[#4D397A] rounded-full`}
      >
        {isDesktop ? 'Connect Wallet' : <WalletIcon className='w-6 h-6' />}
      </button>
      <PreferredExtensionModal isOpen={walletModalIsOpen} onClose={handleWalletModalOnClose} />
    </>
  )
}
