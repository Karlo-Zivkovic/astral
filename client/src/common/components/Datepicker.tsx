import React, { FC, useState } from 'react'
import Datepicker from 'tailwind-datepicker-react'
import { IDatePickerProps } from 'tailwind-datepicker-react/types/Components/DatePicker'

const options = {
  title: 'Demo Title',
  autoHide: true,
  todayBtn: false,
  clearBtn: true,
  maxDate: new Date('2030-01-01'),
  minDate: new Date('1950-01-01'),
  theme: {
    background: 'bg-gray-700 dark:bg-gray-800',
    todayBtn: '',
    clearBtn: '',
    icons: '',
    text: '',
    disabledText: 'bg-red-500',
    input: '',
    inputIcon: '',
    selected: '',
  },
  icons: {
    // () => ReactElement | JSX.Element
    prev: () => <span>Previous</span>,
    next: () => <span>Next</span>,
  },
  datepickerClassNames: 'top-12',
  defaultDate: new Date('2022-01-01'),
  language: 'en',
}

const BasicDatepicker: FC<IDatePickerProps> = (props) => {
  return <Datepicker options={options} {...props} />
}

export default BasicDatepicker
