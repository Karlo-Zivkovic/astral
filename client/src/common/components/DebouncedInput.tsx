import _ from 'lodash'
import React, { useCallback, useEffect, useState } from 'react'

interface DebouncedInputProps
  extends Omit<React.InputHTMLAttributes<HTMLInputElement>, 'onChange'> {
  value: string | number
  onChange: (value: string | number) => void
  delay?: number
}

const DebouncedInput: React.FC<DebouncedInputProps> = ({
  value,
  onChange,
  delay = 300,
  ...props
}) => {
  const [inputValue, setInputValue] = useState(value)

  // Update local state when value prop changes
  useEffect(() => {
    setInputValue(value)
  }, [value])

  // Create a debounced version of the passed onChange function
  // eslint-disable-next-line react-hooks/exhaustive-deps
  const debouncedOnChange = useCallback(
    _.debounce((value: string | number) => {
      onChange(value)
    }, delay),
    [onChange, delay],
  )

  // Update debounced value when input changes
  useEffect(() => {
    if (inputValue !== value) {
      debouncedOnChange(inputValue)
    }
    return () => {
      debouncedOnChange.cancel()
    }
  }, [inputValue, debouncedOnChange, value])

  return <input {...props} value={inputValue} onChange={(e) => setInputValue(e.target.value)} />
}

export default DebouncedInput
