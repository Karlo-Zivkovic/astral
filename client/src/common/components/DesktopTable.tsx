import { ChevronDownIcon, ChevronUpDownIcon, ChevronUpIcon } from '@heroicons/react/20/solid'
import { Table, flexRender } from '@tanstack/react-table'

interface TableProps<T extends object> {
  table: Table<T>
}

const DesktopTable = <T extends object>({ table }: TableProps<T>) => (
  <table className="min-w-max w-full table-auto font-['Montserrat'] bg-white rounded-[20px] dark:bg-gradient-to-r dark:from-[#4141B3] dark:via-[#6B5ACF] dark:to-[#896BD2] dark:border-none">
    <thead className='text-[#857EC2] text-sm dark:text-white/75 border-b border-gray-200'>
      {table.getHeaderGroups().map((headerGroup) => (
        <tr key={headerGroup.id}>
          {headerGroup.headers.map((header) => (
            <th
              key={header.id}
              {...(header.column.getCanSort()
                ? { onClick: header.column.getToggleSortingHandler() }
                : {})}
              className='px-6 py-4 text-sm font-normal text-start'
            >
              <div className='flex gap-2 justify-items-center align-middle'>
                {header.isPlaceholder
                  ? null
                  : flexRender(header.column.columnDef.header, header.getContext())}
                {header.column.getIsSorted() === 'asc' ? (
                  <span className='cursor-pointer relative inline-flex items-center rounded-full text-sm font-medium text-[#DE67E4]  focus:z-20 mr-[14px] dark:text-white  dark:border-none'>
                    <span className='sr-only'>Up</span>
                    <ChevronUpIcon className='h-5 w-5' aria-hidden='true' />
                  </span>
                ) : header.column.getIsSorted() === 'desc' ? (
                  <span className='cursor-pointer relative inline-flex items-center rounded-full text-sm font-medium text-[#DE67E4] focus:z-20 mr-[14px] dark:text-white  dark:border-none'>
                    <span className='sr-only'>Up</span>
                    <ChevronDownIcon className='h-5 w-5' aria-hidden='true' />
                  </span>
                ) : (
                  <span className='cursor-pointer relative inline-flex items-center rounded-full text-sm font-medium text-[#DE67E4]  focus:z-20 mr-[14px] dark:text-white  dark:border-none'>
                    <span className='sr-only'>Up</span>
                    <ChevronUpDownIcon className='h-5 w-5' aria-hidden='true' />
                  </span>
                )}
              </div>
            </th>
          ))}
        </tr>
      ))}
    </thead>
    <tbody className='text-gray-600 text-sm font-light dark:text-white'>
      {table.getRowModel().rows.map((row, index) => (
        <tr
          key={row.id}
          className={`hover:bg-gray-100 dark:hover:bg-transparent/10 ${
            index !== table.getRowModel().rows.length - 1 && 'border-y border-gray-200'
          } `}
        >
          {row.getVisibleCells().map((cell) => (
            <td className='whitespace-nowrap px-6 py-4 text-sm font-light ' key={cell.id}>
              {flexRender(cell.column.columnDef.cell, cell.getContext())}
            </td>
          ))}
        </tr>
      ))}
    </tbody>
  </table>
)

export default DesktopTable
