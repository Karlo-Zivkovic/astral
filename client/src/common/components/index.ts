import Accordion from 'common/components/Accordion'
import Arguments from 'common/components/Arguments'
import ArrowButton from 'common/components/ArrowButton'
import CopyButton from 'common/components/CopyButton'
import Dropdown from 'common/components/Dropdown'
import ErrorFallback from 'common/components/ErrorFallback'
import EventCard from 'common/components/EventCard'
import ExtrinsicCard from 'common/components/ExtrinsicCard'
import HeaderBlockLink from 'common/components/HeaderBlockLink'
import { List, ListItem, StyledListItem } from 'common/components/List'
import MobileCard from 'common/components/MobileCard'
import PageTabs from 'common/components/PageTabs'
import Pagination from 'common/components/Pagination'
import SearchBar from 'common/components/SearchBar'
import Spinner from 'common/components/Spinner'
import StatItem from 'common/components/StatItem'
import StatusIcon from 'common/components/StatusIcon'
import Table, { Column } from 'common/components/Table'
import TableLoadingSkeleton from 'common/components/TableLoadingSkeleton'
import { Tab, Tabs } from 'common/components/Tabs'
import Tooltip from './Tooltip'

export {
  Accordion,
  Arguments,
  ArrowButton,
  CopyButton,
  Dropdown,
  ErrorFallback,
  EventCard,
  ExtrinsicCard,
  HeaderBlockLink,
  List,
  ListItem,
  MobileCard,
  PageTabs,
  Pagination,
  SearchBar,
  Spinner,
  StatItem,
  StatusIcon,
  StyledListItem,
  Tab,
  Table,
  TableLoadingSkeleton,
  Tabs,
  Tooltip,
}

export type { Column }
