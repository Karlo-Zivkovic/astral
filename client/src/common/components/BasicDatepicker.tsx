import React, { FC, useState } from 'react'
import Datepicker from 'tailwind-datepicker-react'
import dayjs from 'dayjs'
import utc from 'dayjs/plugin/utc'
import { CalendarIcon } from '@heroicons/react/24/outline'

dayjs.extend(utc)

type Props = {
  value?: string | undefined
  onChange?: (selectedDate: Date) => void
}

const options = {
  autoHide: true,
  todayBtn: true,
  clearBtn: true,
  theme: {
    background: 'bg-white dark:bg-[#1E254E] dark:text-white',
    todayBtn: '',
    clearBtn: '',
    icons: '',
    text: '',
    disabledText: 'text-gray-500/75 dark:text-gray-400/75',
    input: 'py-3 w-full rounded-[20px] bg-white dark:bg-[#1E254E] dark:text-white',
    inputIcon: '',
    selected: '',
  },
}

const BasicDatepicker: FC<Props> = ({ onChange, value }) => {
  const [show, setShow] = useState<boolean>(false)

  const handleClose = (state: boolean) => {
    setShow(state)
  }

  return (
    <Datepicker options={options} onChange={onChange} show={show} setShow={handleClose}>
      <div className='relative'>
        <span className='absolute flex py-[22px] left-3 items-center h-5 w-5 text-[#282929] dark:text-white'>
          <CalendarIcon />
        </span>
        <input
          type='text'
          placeholder='Select Date'
          value={value ? dayjs(value).utc().format('MMMM DD, YYYY') : ''}
          onFocus={() => setShow(!show)}
          className='px-10 py-3 w-full rounded-[42px] bg-white dark:bg-[#1E254E] dark:text-white border-transparent focus:border-gray-500 focus:bg-white focus:ring-0 text-sm'
          readOnly
        />
      </div>
    </Datepicker>
  )
}

export default BasicDatepicker
