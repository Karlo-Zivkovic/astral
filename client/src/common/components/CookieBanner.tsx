import { useSafeLocalStorage } from 'common/hooks/useSafeLocalStorage'
import { EXTERNAL_ROUTES } from 'common/routes'

const CookieBanner = () => {
  const [isVisible, setIsVisible] = useSafeLocalStorage('cookie-banner', true)

  if (!isVisible) {
    return <></>
  }

  const handleAcceptance = () => {
    setIsVisible(false)
  }

  return (
    <div className='w-full flex items-end'>
      <div className='w-full flex justify-between align-middle text-gray-600 text-sm font-medium dark:text-white'>
        <p>
          By clicking <b>“Accept Cookies”</b>, you agree to the storing of cookies on your device to
          enhance site navigation, analyze site usage, and assist in our marketing efforts.
          <br /> View our{' '}
          <a
            href={EXTERNAL_ROUTES.subspacePrivacyPolicy}
            target='_blank'
            rel='noreferrer'
            className=' text-[#DE67E4] '
          >
            Privacy Policy
          </a>{' '}
          for more information.
        </p>
        <button
          className='w-32 py-2 bg-white dark:bg-[#1E254E] hover:bg-gray-200 text-gray-800 dark:text-white text-sm font-medium rounded-[20px]'
          onClick={handleAcceptance}
        >
          Accept Cookies
        </button>
      </div>
    </div>
  )
}

export default CookieBanner
