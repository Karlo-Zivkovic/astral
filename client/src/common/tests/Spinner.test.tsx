import { render } from '@testing-library/react'

// common - table
import { Spinner } from 'common/components'

describe('Search bar', () => {
  it('renders properly', async () => {
    const { asFragment } = render(<Spinner />)

    expect(asFragment()).toMatchSnapshot()
  })
})
