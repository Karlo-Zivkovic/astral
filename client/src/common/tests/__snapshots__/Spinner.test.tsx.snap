// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Search bar renders properly 1`] = `
<DocumentFragment>
  <div
    class=" w-full min-h-screen flex justify-center items-center"
    data-testid="testLoading"
  >
    <div
      class="flex min-h-screen w-full items-center justify-center "
    >
      <div
        class="flex h-14 w-14 items-center justify-center rounded-full bg-gradient-to-tr from-[#ABCFEF] via-[#929EEA] to-[#91D3A0] animate-spin"
      >
        <div
          class="h-9 w-9 rounded-full background-gradient"
        />
      </div>
    </div>
  </div>
</DocumentFragment>
`;
